runtime/src/tiered_storage/writer.rs

```diff
@@ -0,0 +1,86 @@
+//! docs/src/proposals/append-vec-storage.md
+
+use {
+    crate::{
+        account_storage::meta::{StorableAccountsWithHashesAndWriteVersions, StoredAccountInfo},
+        storable_accounts::StorableAccounts,
+        tiered_storage::{
+            error::TieredStorageError,
+            file::TieredStorageFile,
+            footer::{AccountMetaFormat, TieredStorageFooter},
+            hot::HotAccountMeta,
+            meta::TieredAccountMeta,
+            TieredStorageFormat, TieredStorageResult,
+        },
+    },
+    solana_sdk::{account::ReadableAccount, hash::Hash},
+    std::{borrow::Borrow, path::Path},
+};
+
+#[derive(Debug)]
+pub struct TieredStorageWriter<'format> {
+    storage: TieredStorageFile,
+    format: &'format TieredStorageFormat,
+}
+
+impl<'format> TieredStorageWriter<'format> {
+    pub fn new(
+        file_path: impl AsRef<Path>,
+        format: &'format TieredStorageFormat,
+    ) -> TieredStorageResult<Self> {
+        Ok(Self {
+            storage: TieredStorageFile::new_writable(file_path)?,
+            format,
+        })
+    }
+
+    #[inline]
+    fn write_accounts_impl<
+        'a,
+        'b,
+        T: ReadableAccount + Sync,
+        U: StorableAccounts<'a, T>,
+        V: Borrow<Hash>,
+        W: TieredAccountMeta,
```

Comment:
Note that the `write_accounts_impl` has one more generic than `write_accounts()`: TieredAccountMeta.